app/features/counter/Counter.js

```diff
@@ -36,23 +36,23 @@ export default class Counter extends Component<Props> {
         <div className={styles.btnGroup}>
           <button
             className={styles.btn}
-            onClick={increment}
+            onClick={() => increment()}
```

Comment:
Probably not a big deal, but calling functions in his way create new references every time the component re-renders.

Reply:
I agree with. @johncantrell97 Is this a necessary change?

Reply:
It is actually rather necessary.

By default, the action creators generated by Redux Toolkit accept a single argument, which becomes `action.payload`.  `onClick` passes a React event object to the click handler.  So, `onClick={increment}` will end up passing the event object as the action payload.

This becomes an issue because RTK also adds a "serializability check" middleware to the store setup by default, warning about any non-serializable values in actions and state.  The event objects aren't serializable, so RTK will warn whenever this is clicked.

Reply:
Thanks for the clear explanation. Makes a lot of sense.

Reply:
Thanks for explaining this @markerikson.